packages/wasm-dpp/src/identity/identity_factory.rs

```diff
@@ -0,0 +1,271 @@
+        asset_lock_transaction: Vec<u8>,
+        output_index: u32,
+    ) -> Result<InstantAssetLockProofWasm, JsValue> {
+        let instant_lock: InstantLock =
+            consensus::deserialize(&instant_lock).map_err(|e| e.to_string())?;
+
+        let asset_lock_transaction: Transaction =
+            consensus::deserialize(&asset_lock_transaction).map_err(|e| e.to_string())?;
+
+        Ok(IdentityFactory::<BlsAdapter>::create_instant_lock_proof(
+            instant_lock,
+            asset_lock_transaction,
+            output_index,
+        )
+        .into())
+    }
+
+    #[wasm_bindgen(js_name=createChainAssetLockProof)]
+    pub fn create_chain_asset_lock_proof(
+        &self,
+        core_chain_locked_height: u32,
+        out_point: Vec<u8>,
+    ) -> Result<ChainAssetLockProofWasm, JsValue> {
+        let out_point: [u8; 36] = out_point.try_into().map_err(|_| {
+            RustConversionError::Error(String::from("outPoint must be a 36 byte array"))
+                .to_js_value()
+        })?;
+
+        Ok(
+            IdentityFactory::<BlsAdapter>::create_chain_asset_lock_proof(
+                core_chain_locked_height,
+                out_point,
+            )
+            .into(),
+        )
+    }
+
+    #[wasm_bindgen(js_name=createIdentityCreateTransition)]
+    pub fn create_identity_create_transition(
+        &self,
+        identity: &IdentityWasm,
+    ) -> Result<IdentityCreateTransitionWasm, JsValue> {
+        self.0
+            .create_identity_create_transition(Identity::from(identity.to_owned()))
+            .map(Into::into)
+            .map_err(from_dpp_err)
+    }
+
+    #[wasm_bindgen(js_name=createIdentityTopUpTransition)]
+    pub fn create_identity_topup_transition(
+        &self,
+        identity_id: &IdentifierWrapper,
+        asset_lock_proof: &JsValue,
+    ) -> Result<IdentityTopUpTransitionWasm, JsValue> {
+        let asset_lock_proof = create_asset_lock_proof_from_wasm_instance(asset_lock_proof)?;
+
+        self.0
+            .create_identity_topup_transition(identity_id.to_owned().into(), asset_lock_proof)
+            .map(Into::into)
+            .map_err(from_dpp_err)
+    }
+
+    #[wasm_bindgen(js_name=createIdentityUpdateTransition)]
+    pub fn create_identity_update_transition(
+        &self,
+        identity: &IdentityWasm,
+        public_keys: &JsValue,
+    ) -> Result<IdentityUpdateTransitionWasm, JsValue> {
+        let mut add_public_keys = None;
+
+        if js_sys::Reflect::has(public_keys, &"add".into()).unwrap_or(false) {
+            let raw_add_public_keys = js_sys::Reflect::get(public_keys, &"add".into()).unwrap();
+
+            let add_public_keys_array: &js_sys::Array = raw_add_public_keys
+                .dyn_ref::<js_sys::Array>()
+                .ok_or_else(|| {
+                    RustConversionError::Error(String::from("public keys to add must be array"))
+                        .to_js_value()
+                })?;
```

Comment:
NIT:  extraction array from `add` and `disable`. seems duplicates. Would be possible to extract the functionality to one function?

Reply:
Fixed